perceval/backends/core/groupsio.py

```diff
@@ -162,18 +170,53 @@ def fetch(self):
         if not os.path.exists(self.dirpath):
             os.makedirs(self.dirpath)
 
-        url = urijoin(GROUPSIO_API_URL, 'downloadarchives')
-        payload = {'group_name': self.group_name}
-        auth = HTTPBasicAuth(self.api_token, '')
+        group_id = self.__find_group_id()
 
+        url = urijoin(GROUPSIO_API_URL, self.DOWNLOAD_ARCHIVES)
+        payload = {'group_id': group_id}
         filepath = os.path.join(self.dirpath, MBOX_FILE)
-        success = self._download_archive(url, payload, auth, filepath)
+        success = self._download_archive(url, payload, filepath)
 
         return success
 
-    def _download_archive(self, url, payload, auth, filepath):
+    def subscriptions(self, per_page=PER_PAGE):
+        """Fetch the groupsio paginated subscriptions for a given token
+
+        :param per_page: number of subscriptions per page
+
+        :returns: an iterator of subscriptions
+        """
+        url = urijoin(GROUPSIO_API_URL, self.GET_SUBSCRIPTIONS)
+        logger.debug("Get groupsio paginated subscriptions from " + url)
+
+        payload = {
+            "limit": per_page
+        }
+
+        r = self.__fetch(url, payload)
```

Comment:
This whole logic can be merged into a single `while` loop instead of repeating the code twice.